### ISSUE TITLE
inclusion of logging.properties in the war causing JBoss failures

### ISSUE THREAD

**Issue by mrgmatthews** [spring-projects/spring-boot#7257]
logging.properties is a reserved name. So when spring jar is bundled within the deployable war it tries to use that.
Logs:

```
2016-10-31 09:14:40,762 DEBUG [org.jboss.as.logging] (MSC service thread 1-6)  | Found logging configuration file: "/content/XXX.war/WEB-INF/lib/spring-boot-1.4.1.RELEASE.jar/org/springframework/boot/logging/java/logging.properties"
2016-10-31 09:14:40,763 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-6)  | MSC00001: Failed to start service jboss.deployment.unit."XXX.war".POST_MODULE: org.jboss.msc.service.StartException in service jboss.deployment.unit."XXX.war".POST_MODULE: JBAS018733: Failed to process phase POST_MODULE of deployment "XXX.war"
        at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:123) [jboss-as-server-7.1.2.Final.jar:7.1.2.Final]
        at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1811) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
        at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1746) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_72]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_72]
        at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_72]
Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: JBAS011555: Failed to configure logging using 'logging.properties' configuration file.
        at org.jboss.as.logging.LoggingConfigurationProcessor.deploy(LoggingConfigurationProcessor.java:125)
        at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:116) [jboss-as-server-7.1.2.Final.jar:7.1.2.Final]
        ... 5 more
Caused by: java.lang.IllegalArgumentException: className is null
        at org.jboss.logmanager.config.AbstractPropertyConfiguration.<init>(AbstractPropertyConfiguration.java:52) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.logmanager.config.HandlerConfigurationImpl.<init>(HandlerConfigurationImpl.java:54) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.logmanager.config.LogContextConfigurationImpl.addHandlerConfiguration(LogContextConfigurationImpl.java:138) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.logmanager.PropertyConfigurator.configureHandler(PropertyConfigurator.java:477) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.logmanager.PropertyConfigurator.configure(PropertyConfigurator.java:379) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.logmanager.PropertyConfigurator.configure(PropertyConfigurator.java:92) [jboss-logmanager-1.3.1.Final.jar:1.3.1.Final]
        at org.jboss.as.logging.LoggingConfigurationProcessor.deploy(LoggingConfigurationProcessor.java:122)
        ... 6 more
```

Would like the name to be changed. So JavaLoggingSystem  would also need to be modified

**Comment by wilkinsona** [comment]
What version of JBoss are you using? We have a deployment test that uses Wildfly 8.2.0.Final that isn't affected by this problem.

**Comment by mrgmatthews** [comment]
jboss-as-server-7.1.2.Final

**Comment by wilkinsona** [comment]
As far as I can tell, this is a JBoss problem. The [documentation](https://docs.jboss.org/author/display/AS72/Logging+Configuration) states:

> Per-deployment logging allows you to add a logging configuration file to your deployment and have the logging for that deployment configured according to the configuration file. In an EAR the configuration should be in the META-INF directory. In a WAR or JAR deployment the configuration file can be in either the META-INF or WEB-INF/classes directories.

According to that, it shouldn't be looking at our `logging.properties` file in `org/springframework/boot/logging/java/` inside a jar in `WEB-INF/lib`, and it certainly shouldn't be failing because it's not in the required format for JBoss's per-deployment logging.

Unfortunately, we can't make a breaking change to `JavaLoggingSystem` just to appease JBoss. Assuming you have to stick with JBoss 7.1, your best bet is probably to disable per-deployment logging configuration. You can do that system wide using a system property. Hopefully there's a way to disable it per-deployment as well.

**Comment by mrgmatthews** [comment]
The answer to this makes me sad

**Comment by wilkinsona** [comment]
I sympathise, but I really don't think we can do anything. Given the constraint that we can't make a breaking change to `JavaLoggingSystem`, what do you suggest that we do instead?

**Comment by mrgmatthews** [comment]
is renaming the logging.properties file to java-logging.properties that big of a breaking change?

**Comment by wilkinsona** [comment]
Yes, I think it is. The problem is that it'll affect anyone using `JavaLoggingSystem` just to fix a problem with JBoss. I find that very hard to justify particularly as a workaround is available for 7.1, and more recent versions don't have the problem as JBoss's behaviour has been corrected.
